presto-parser/src/main/java/com/facebook/presto/sql/ExpressionFormatter.java

```diff
@@ -566,6 +573,53 @@ static String formatSortItems(List<SortItem> sortItems, boolean unmangleNames)
                 .iterator());
     }
 
+    static String formatGroupBy(List<GroupingElement> groupingElements)
+    {
+        ImmutableList.Builder<String> resultStrings = ImmutableList.builder();
+
+        for (GroupingElement groupingElement : groupingElements) {
+            String result = "";
+            if (groupingElement instanceof SimpleGroupBy) {
+                List<Expression> columns = ((SimpleGroupBy) groupingElement).getColumnExpressions();
+                if (columns.size() == 1) {
+                    result = formatExpression(getOnlyElement(columns));
+                }
+                else {
+                    result = format("(%s)", Joiner.on(", ").join(columns.stream()
+                            .map(ExpressionFormatter::formatExpression)
+                            .iterator()));
+                }
+            }
+            else if (groupingElement instanceof GroupingSets) {
+                result = format("GROUPING SETS (%s)", Joiner.on(", ").join(
+                        groupingElement.enumerateGroupingSets().stream()
+                                .map(ExpressionFormatter::formatGroupingSet)
+                                .iterator()));
+            }
+            else if (groupingElement instanceof Cube) {
+                result = format("CUBE (%s)", Joiner.on(", ").join(
+                        ((Cube) groupingElement).getColumns().stream()
+                                .map(QualifiedName::toString)
```

Comment:
This is not needed. `Joiner.join` calls `toString` implicitly.

So this can just be: `format("CUBE (%s)", Joiner.on(", ").join((Cube) groupingElement).getColumns()))`